dev/how-to/maintain/tidb-binlog.md

```diff
@@ -6,41 +6,57 @@ category: reference
++ how to manage Binlog clusters by using the binlogctl tool or by directly performing SQL operations in TiDB
+
 ## Pump or Drainer state
 
 Pump or Drainer state description:
 
-* `online`: running normally.
-* `pausing`: in the pausing process. It turns into this state after you use `kill` or press Ctrl + C to exit from the process. When Pump or Drainer exits all internal threads in safe, it becomes `paused`.
-* `paused`: has been stopped. While Pump is in this state, it rejects the request of writing binlog into it and does not provide the binlog for Drainer any more. When Drainer is in this state, it does not replicate data to the downstream. After Pump and Drainer exit normally from all the threads, they switch the state to `paused` and then exits from the process.
-* `closing`: in the offline process. The `binlogctl` is used to get Pump or Drainer offline and Pump or Drainer is in this state before the process exits. In this state, Pump does not accept new requests of writing binlog into it and waits for all the binlog data to be used up by Drainer.
-* `offline`: becomes offline. After Pump sends all the binlog data that it saves to Drainer, its state is switched to `offline`.
+* `online`: running normally
+* `pausing`: in the pausing process
+* `paused`: has been stopped
+* `closing`: in the offline process
+* `offline`: becomes offline
 
 > **Note:**
 >
-> * When Pump or Drainer is `pausing` or `paused`, the data replication is interrupted.
-> * Pump and Drainer have several states, including `online`, `paused`, and `offline`. If you press Ctrl + C or `kill` the process, both Pump and Drainer become `pausing` then finally turn to `paused` . There is no need for Pump to send all the binlog data to Drainer before it become `paused` while Pump need to send all the binlog data to Drainer before it become `offline` . If you need to exit from Pump for a long period of time (or are permanently removing Pump from the cluster), use `binlogctl` to make Pump offline. The same goes for Drainer.
-> * When Pump is `closing`, you need to guarantee that all the data has been consumed by all the Drainers that are not `offline`. So before making Pump offline, you need to guarantee all the Drainers are `online`. Otherwise, Pump cannot get offline normally.
-> * The binlog data that Pump saves is processed by GC only when it has been consumed by all the Drainers that are not `offline`.
-> * Close Drainer only when it will not be used any more.
+> The state information of Pump or Drainer node is maintained internally in Pump or Drainer and regularly updated to the Placement Driver (PD).
+
+## Starting and exiting Pump or Drainer process
+
+### Pump
+
+* Starting: When started, the Pump nodes notifies all Drainer nodes in the `online` state. If the notification is successful, the Pump node sets its state to `online`. Otherwise, the node reports an error, sets its state to `paused` and exits the process.
```

Comment:
```suggestion
* Starting: When started, the Pump node notifies all Drainer nodes in the `online` state. If the notification is successful, the Pump node sets its state to `online`. Otherwise, the Pump node reports an error, sets its state to `paused` and exits the process.
```